File: integerpartition.js
function printarray(a,number) //Function of Printing Array
{
  for(let i=0;i<number;i++)
      a.join(''); 
  console.log(a);
}
function integerpartition(number) //Function of Integer Partition
{
  let a=[]; 
  let last=0;
  a[last]=number;
  while(true)
    {
      printarray(a,last+1);
      let left=0;
      while(last>=0 && a[last]==1)
        {
          left+=a[last];
          last--;
        }
      if(last<0)
        return;
      a[last]--;
      left++;
      while(left>a[last])
        {
          a[last+1]=a[last];
          left=left-a[last];
          last++;
        }
      a[last+1]=left;
      last++;
    }
}
integerpartition(5); //Call Integer Partition Function

File: IntegerPartition.scala

/*Program in Scala for Integer Partition*/

object Main {
	def main(args: Array[String])
	{
        val num = 8
        //Defining matrix size
        var mat = Array.ofDim[Int](num+1,num+1)
        //Fill the first row to 0
        for(i<-1 to num)
            mat(0)(i) = 0
        //Fill the first column to 1
        for(i<-0 to num)
            mat(i)(0) = 1
        //Finding the combinations of a number that it can be divided into
        for(x<-1 to num)
            for(y<-1 to num)
            {
                if(x > y)
                    mat(x)(y) = mat(x-1)(y)
                else
                {
                    val comboWithout = mat(x-1)(y)
                    val comboWith = mat(x)(y-x)
                    mat(x)(y) = comboWithout + comboWith
                }
            }
        //Displaying the ways
        println(num+" can be divided into "+mat(num)(num)+" ways")
	}
}